pkg/util/util.go

```diff
@@ -90,6 +90,56 @@ type ValidationError struct {
 	Msg string
 }
 
+// Options is a struct holding complexity params: length, upper, lower, numbers and symbols
+type Options struct {
+	Length  int
+	Upper   bool
+	Lower   bool
+	Numbers bool
+	Symbols bool
+}
+
+// ValidateComplexity checks if a string meets specified complexity requirements.
+// It takes a string 'str' and an 'options' struct containing criteria such as length,
+// uppercase, lowercase, numbers, and symbols. Returns true if the string meets the
+// criteria, otherwise false.
+func ValidateComplexity(str string, options Options) bool {
```

Comment:
Please notice the changes in core in PR [noobaa/noobaa-core/#8008](https://github.com/noobaa/noobaa-core/pull/8008), see here:
https://github.com/noobaa/noobaa-core/blob/56e17689c9937ee91f5516b8ed73f1ceb1eee6e5/src/util/string_utils.js#L11-L12

Reply:
Thanks, I will update core PR accordingly.